amplitude.js

```diff
@@ -801,18 +830,41 @@ AmplitudeClient.prototype._setInStorage = function _setInStorage(storage, key, v
 };
 
 /**
+ * Helper function to remove values in storage
+ * Storage argument allows for localStoraoge and sessionStoraoge
+ * @private
+ */
+AmplitudeClient.prototype._removeFromStorage = function _removeFromStorage(storage, key) {
+  storage.removeItem(key + this._storageSuffix);
+};
+
+/**
  * cookieData (deviceId, userId, optOut, sessionId, lastEventTime, eventId, identifyId, sequenceNumber)
  * can be stored in many different places (localStorage, cookie, etc).
  * Need to unify all sources into one place with a one-time upgrade/migration.
  * @private
  */
 var _upgradeCookeData = function _upgradeCookeData(scope) {
-  // skip if migration already happened
-  var cookieData = scope.cookieStorage.get(scope.options.cookieName);
+  // check if already migrated to new instance keys with scope
+  var scopedKey = scope.options.cookieName + scope._apiKeySuffix + scope._storageSuffix;
```

Comment:
since the key now consists of amplitude + domain + apiKey + storage scope, how long does the key end up being?